www/js/splash/notifScheduler.ts

```diff
@@ -0,0 +1,286 @@
+            })
+        }
+    })
+}
+
+//get scheduled notifications from cordova plugin and format them
+const getNotifs = function() {
+    return new Promise((resolve, reject) => {
+        window['cordova'].plugins.notification.local.getScheduled((notifs) => {
+            if (!notifs?.length){
+                console.log("there are no notifications");
+                resolve([]); //if none, return empty array
+            }
+            
+            const notifSubset = notifs.slice(0, 5); //prevent near-infinite listing
+            let scheduledNotifs = [];
+            scheduledNotifs = notifSubset.map((n) => {
+                const time = DateTime.fromMillis(n.trigger.at).toFormat('t');
+                const date = DateTime.fromMillis(n.trigger.at).toFormat('DDD');
+                return {
+                    key: date,
+                    val: time
+                }
+            });
+            resolve(scheduledNotifs);
+         });
+    })
+}
+
+// schedules the notifications using the cordova plugin
+const scheduleNotifs = (scheme, notifTimes) => {
+    return new Promise<void>((rs) => {
+        isScheduling = true;
+        const localeCode = i18next.resolvedLanguage;
+        console.error("notifTimes: ", notifTimes, " - type: ", typeof(notifTimes));
+        const nots = notifTimes.map((n) => {
+            console.error("n: ", n, " - type: ", typeof(n));
+            const nDate = n.toDate();
+            const seconds = nDate.getTime() / 1000;
+            return {
+                id: seconds,
+                title: scheme.title[localeCode],
+                text: scheme.text[localeCode],
+                trigger: {at: nDate},
+                // actions: 'reminder-actions',
+                // data: {
+                //     action: {
+                //         redirectTo: 'root.main.control',
+                //         redirectParams: {
+                //             openTimeOfDayPicker: true
+                //         }
+                //     }
+                // }
+            }
+        });
+        window['cordova'].plugins.notification.local.cancelAll(() => {
+            debugGetScheduled("After cancelling");
+            window['cordova'].plugins.notification.local.schedule(nots, () => {
+                debugGetScheduled("After scheduling");
+                isScheduling = false;
+                rs(); //scheduling promise resolved here
+            });
+        });
+    });
+}
+
+// determines when notifications are needed, and schedules them if not already scheduled
+const update = async (reminderSchemes) => {
+    const { reminder_assignment,
+            reminder_join_date,
+            reminder_time_of_day} = await getReminderPrefs(reminderSchemes);
+    var scheme = {};
+    try {
+        scheme = reminderSchemes[reminder_assignment];
+    } catch (e) {
+        console.log("ERROR: Could not find reminder scheme for assignment " + reminderSchemes + " - " + reminder_assignment);
+    }
```

Comment:
Appreciate the addition of this catch block - but instead of `console.log` let's `displayErrorMsg` (from logger.ts)